src/entities/histogram/histogramImpl.js

```diff
@@ -0,0 +1,385 @@
+   * Will update _priorMinuteBinsList if the minute has passed.
+   * @param currMinuteMillis
+   * @returns {MinuteBin} MinuteBin of current minute.
+   */
+  getOrUpdateCurrentBin(currMinuteMillis) {
+    if (this._currentMinuteBin.minuteMillis === currMinuteMillis) {
+      return this._currentMinuteBin;
+    }
+    if (this._priorMinuteBinsList.length > this._MAX_BINS) {
+      this._priorMinuteBinsList.shift();
+    }
+    this._priorMinuteBinsList.push(this._currentMinuteBin);
+    this._currentMinuteBin = new MinuteBin(this._ACCURACY, currMinuteMillis);
+    return this._currentMinuteBin;
+  }
+
+  /**
+   * Return newly-updated _priorMinuteBinsList.
+   * @returns {[]|Array}
+   */
+  getPriorMinuteBinsList() {
+    this.getOrUpdateCurrentBin(this.currentMinuteMillis());
+    return this._priorMinuteBinsList;
+  }
+
+  /**
+   * Return the standard deviation of the values in the distribution.
+   */
+  stdDev() {
+    let mean = this.getMean();
+    let varianceSum = 0,
+      count = 0;
+    for (let minuteBin of this.getPriorMinuteBinsList()) {
+      let centroids = minuteBin.getCentroids();
+      count += centroids.reduce((accumulator, c) => accumulator + c.n, 0);
+      varianceSum += centroids.reduce(
+        (accumulator, c) => accumulator + c.n * (c.mean - mean) ** 2,
+        0
+      );
+    }
+    let variance = count === 0 ? 0 : varianceSum / count;
+    return Math.sqrt(variance);
+  }
+
+  /**
+   * Aggregate all the minute bins prior to the current minute.
+   * Returns a list of the distributions held within each bin.
+   * Note that invoking this method will also clear all data from the
+   * aggregated bins, thereby changing the state of the system and
+   * preventing data from being flushed more than once.
+   * @returns {Distribution[]}
+   */
+  flushDistributions() {
+    let distributions = [];
+    this.getOrUpdateCurrentBin(this.currentMinuteMillis());
+    this._priorMinuteBinsList.forEach(minuteBin =>
+      distributions.push(minuteBin.toDistribution())
+    );
+    this._priorMinuteBinsList = [];
+    return distributions;
+  }
+
+  /**
+   * Get the number of values in the distribution.
+   * @return {number}
+   */
+  getCount() {
+    return this.getPriorMinuteBinsList().reduce(
+      (accumulator, minuteBin) => accumulator + minuteBin.dist.n,
+      0
+    );
+  }
+
+  /**
+   * Get the sum of the values in the distribution.
+   */
+  getSum() {
+    let res = 0;
+    this.getPriorMinuteBinsList().forEach(minuteBin => {
+      res += minuteBin
+        .getCentroids()
+        .reduce((accumulator, c) => accumulator + c.mean * c.n, 0);
+    });
+    return res;
+  }
+
+  /**
+   * Return a statistical of the histogram distribution.
+   * @return {Snapshot} Snapshot of Histogram
+   */
+  getSnapshot() {
+    let newDigest = new TDigest(1 / this._ACCURACY);
+    for (let minuteBin of this.getPriorMinuteBinsList()) {
+      minuteBin.getCentroids().forEach(c => newDigest.push(c.mean, c.n));
```

Comment:
This seems like it would be very memory-intensive for high cardinality datasets. Ex: mean from 1..100K, n = 1 for all of them. 10 buckets means 10x memory consumption.

Reply:
Yes I agree that this is an expensive operation for large dataset. just trying to be consistent here with the other SDKs.